pkg/virt-controller/watch/migration_test.go

```diff
@@ -1982,6 +2017,79 @@ var _ = Describe("Migration watcher", func() {
+
+			addMigration(migration)
+			addVirtualMachineInstance(vmi)
+			sourcePod := newSourcePodForVirtualMachine(vmi)
+			addPod(sourcePod)
+
+			controller.Execute()
+
+			testutils.ExpectEvents(recorder, SuccessfulCreatePodReason)
+			expectPodCreation(vmi.Namespace, vmi.UID, migration.UID, 1, 0, 0)
+			updatedSourcePod, err := kubeClient.CoreV1().Pods(vmi.Namespace).Get(context.Background(), sourcePod.Name, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(updatedSourcePod.Annotations).ToNot(HaveKey(deschedulerEvictionInProgressAnnotation))
+		})
+
+		Context("with an evacuation migration", func() {
+			It("should add eviction-in-progress annotation only to source virt-launcher pod", func() {
+				vmi = newVirtualMachine("testvmi", virtv1.Running)
+				migration := newMigration("testmigration", vmi.Name, virtv1.MigrationPending)
+				setEvacuationAnnotation(migration)
+
+				addMigration(migration)
+				addVirtualMachineInstance(vmi)
+				sourcePod := newSourcePodForVirtualMachine(vmi)
+				addPod(sourcePod)
+
+				controller.Execute()
+
+				testutils.ExpectEvents(recorder, SuccessfulCreatePodReason)
+				expectPodCreation(vmi.Namespace, vmi.UID, migration.UID, 1, 0, 0)
+				updatedSourcePod, err := kubeClient.CoreV1().Pods(vmi.Namespace).Get(context.Background(), sourcePod.Name, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				Expect(updatedSourcePod.Annotations).To(HaveKeyWithValue(deschedulerEvictionInProgressAnnotation, ""))
+
+				pods, err := kubeClient.CoreV1().Pods(migration.Namespace).List(context.Background(), metav1.ListOptions{
+					LabelSelector: fmt.Sprintf("%s=%s,%s=%s", virtv1.MigrationJobLabel, string(migration.UID), virtv1.CreatedByLabel, string(vmi.UID)),
+				})
+				Expect(err).ToNot(HaveOccurred())
+				Expect(pods.Items).To(HaveLen(1))
+				targetPod := pods.Items[0]
+				Expect(targetPod.Annotations).ToNot(HaveKey(deschedulerEvictionInProgressAnnotation))
+			})
+
+			It("should remove eviction-in-progress annotation from source virt-launcher pod in case of failure", func() {
+				By("Create a pending migration")
+				vmi = newVirtualMachine("testvmi", virtv1.Running)
+				sourcePod := newSourcePodForVirtualMachine(vmi)
+				sourcePod.Annotations[deschedulerEvictionInProgressAnnotation] = ""
+				migration := newMigration("testmigration", vmi.Name, virtv1.MigrationFailed)
+				migration.Status.MigrationState = &virtv1.VirtualMachineInstanceMigrationState{
+					SourcePod: sourcePod.Name,
+				}
+				setEvacuationAnnotation(migration)
+
+				addMigration(migration)
+				addVirtualMachineInstance(vmi)
+				addPod(sourcePod)
+
+				controller.Execute()
+
+				updatedSourcePod, err := kubeClient.CoreV1().Pods(vmi.Namespace).Get(context.Background(), sourcePod.Name, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				Expect(updatedSourcePod.Annotations).ToNot(HaveKeyWithValue(deschedulerEvictionInProgressAnnotation, ""))
+			})
+		})
+	})
```

Comment:
Maybe also a `descheduler_test.go`?

Reply:
I would leave the test at the controller level, since the descheduler pkg works as an utility.
Also this would save us from a cycle dependency:
```
.-> //pkg/virt-controller/watch/descheduler:go_default_library 
`-- //pkg/virt-controller/watch:go_default_library 
```